semantic_model_generator/data_processing/cte_utils.py

```diff
@@ -77,43 +80,107 @@ def remove_ltable_cte(sql_w_ltable_cte: str) -> str:
+    else:
+        expr_columns = [_get_col_expr(col) for col in columns]
+        cte = f"WITH {logical_table_name(table)} AS (\n"
+        cte += "SELECT \n"
+        cte += ",\n".join(expr_columns) + "\n"
+        cte += f"FROM {fully_qualified_table_name(table.base_table)}"
+        cte += ")"
+        return cte
+
+
+def _generate_non_agg_cte(table: semantic_model_pb2.Table) -> str:
+    """
+    Returns a CTE representing a logical table that selects 'col' columns from 'table' except for aggregation columns.
+    """
+    filtered_cols = [col for col in table.columns if not is_aggregation_expr(col)]
+    return _generate_cte_for(table, filtered_cols)
+
+
+def _convert_to_snowflake_sql(sql: str) -> str:
+    """
+    Converts a given SQL statement to Snowflake SQL syntax using SQLGlot.
+
+    Args:
+    sql (str): The SQL statement to convert.
+
+    Returns:
+    str: The SQL statement in Snowflake syntax.
+    """
+    try:
+        expression = sqlglot.parse_one(sql, dialect=Snowflake)
+    except Exception as e:
+        raise ValueError(
+            f"Unable to parse sql statement.\n Provided sql: {sql}\n. Error: {e}"
         )
 
-    columns = []
-    table_non_agg_column_names = {
-        col.name: col for col in table.columns if not is_aggregation_expr(col)
-    }
-    # If a table has no explicit columns referenced (e.g. for select count(*) from table)
-    # or "*" is in the columns (e.g. for select * from table),
-    # then just add all the columns (excl. ones with aggregation functions in expr).
-    columns.extend(
-        [
-            _get_col_expr(col)
-            for col in table.columns
-            if col.name in table_non_agg_column_names
-        ]
+    return expression.sql()
+
+
+def generate_select(
+    table_in_column_format: semantic_model_pb2.Table, limit: int
+) -> List[str]:
+    """Generate select query for all columns for validation purpose."""
+    # Generate select query for columns without aggregation exprs.
+    non_agg_cte = _generate_non_agg_cte(table_in_column_format)
+    non_agg_sql = (
+        non_agg_cte
+        + f"SELECT * FROM {logical_table_name(table_in_column_format)} LIMIT {limit}"
     )
 
-    cte = f"WITH {logical_table_name(table)} AS (\n"
-    cte += "SELECT \n"
-    cte += ",\n".join(columns) + "\n"
-    cte += f"FROM {fully_qualified_table_name(table.base_table)}"
-    cte += ")"
-    return cte
+    # Generate select query for columns with aggregation exprs.
+    agg_cols = [
+        col for col in table_in_column_format.columns if is_aggregation_expr(col)
+    ]
+    if len(agg_cols) == 0:
+        return [_convert_to_snowflake_sql(non_agg_sql)]
+    else:
+        agg_cte = _generate_cte_for(table_in_column_format, agg_cols)
+        agg_sql = (
+            agg_cte
+            + f"SELECT * FROM {logical_table_name(table_in_column_format)} LIMIT {limit}"
+        )
+    return [_convert_to_snowflake_sql(agg_sql), _convert_to_snowflake_sql(non_agg_sql)]
```

Comment:
do these need to be separate SQLs anymore? seems like you're generating the CTE and the select statement on the CTE the same way.